### PR TITLE
Repair and refine embed block styles

### PR DESCRIPTION
**Pull request by jasmussen**
This is a rebased version of #14648, so props to @m-e-h.

It's a good PR that aims to improve and refine the embed CSS to be less specific and work better. It also fixes some code quality and is overall a good PR. 

Thank you @m-e-h for the PR!

### CODE REVIEW

**Comment by jasmussen** [comment]
Okay, rebased, refreshed, and tested again. The embed block with responsiveness works as it should, but this still makes for better specificity. It also fixes a few typos, and it fixes an issue where a floated embed can collapse into nothingness:

Screenshot 2020-05-11 at 09 57 31

Screenshot 2020-05-11 at 10 20 35

**Comment by ZebulanStanphill** [comment]
I tried to test this PR on gutenberg.run, but trying to embed anything keeps failing there.

Just looking at the code changes, though, they seem like an improvement to me.

**Review by draganescu — APPROVED** [review]
Tested locally. LGTM!

**Comment by jasmussen** [comment]
THANK YOU!

File: packages/block-library/src/embed/style.scss [hunk 1]
```diff
@@ -18,63 +23,62 @@
...
-			padding-top: 100%; // 1 / 1 * 100
-		}
+	.wp-embed-aspect-1-1 .wp-block-embed__wrapper::before {
+		padding-top: 100%; // 1 / 1 * 100
+	}
 
-		&.wp-embed-aspect-9-16 .wp-block-embed__wrapper::before {
-			padding-top: 177.78%; // 16 / 9 * 100
-		}
+	.wp-embed-aspect-9-6 .wp-block-embed__wrapper::before {
```
Comment (westonruter) [inline packages/block-library/src/embed/style.scss:77]: @jasmussen This appears to have been typo, to provide `6` instead of `16`. Fix provided in #25972.

**Pull request merged**